dev-util/unicorn/unicorn-2.0.1.ebuild

```diff
@@ -36,6 +36,13 @@ BDEPEND="virtual/pkgconfig
 
 UNICORN_TARGETS="x86 arm aarch64 riscv mips sparc m68k ppc s390x tricore"
 
+# suppress warning wrt 'implicit function declaration' in config logs due to
```

Comment:
This doesn't really explain it to me. glibc has both of these functions.

My _assumption_ is that you're referring to some silly tests which try to test stuff without an include (or feature test macro), and then if that fails, they include the include or macro. Is that what's happening here? Or are the tests really broken and missing includes/macros entirely?

Reply:
The `configure`-script tries to compile a short code snippet referencing this functions, to determine if the functions are present or not.

```sh
if test "$malloc_trim" != "no" ; then
    cat > $TMPC << EOF
#include <malloc.h>
int main(void) { malloc_trim(0); return 0; }
EOF
    if compile_prog "" "" ; then
        malloc_trim="yes"
    else  
        malloc_trim="no"
    fi    
fi
```

If not, an implicit declaration warning is generated.

What's your recommendation to resolve this issue?

Reply:
The linked bug report was discovered on a musl tinderbox. The "silly tests" were therefore broken on musl, and gcc 14 causes it to start working I guess?

Maybe the solution here is that given gcc 13 is stable, we need to start enforcing `-Werror=implicit-function-declaration`?

Reply:
The comment in the ebuild is overly elaborate while failing to actually get to the point.

> `# suppress warning wrt 'implicit function declaration' in config logs`

We know. The variable "QA_CONFIG_IMPL_DECL_SKIP" will *skip* (suppress) warnings for *implicit decls*. This comment is inherent and implicit in the whole purpose of the variable.

> `# due to auto-detection of some libc functions (bug #906919)`

This is maybe more interesting, but the real question is whether that is okay! So the question we should be asking is not whether the libc functions are being autodetected, but whether the checks are correctly handled (in particular, whether they fail the same on gcc 13). In this case, `malloc_trim(0); return 0` is guaranteed to attempt to link to the malloc_trim symbol, and fail to link if it doesn't work -- and the reason it doesn't work is because the canonical header is correctly included, but won't expose the symbol on libcs where the function doesn't exist.

So all is well, the check is definitely a false positive -- but the comment doesn't really explain why it is a false positive. And that's what we care about -- this QA variable says "someone verified that this is a false positive".